src/Microsoft.AspNet.Mvc.TagHelpers/ImageTagHelper.cs

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.AspNet.Hosting;
+using Microsoft.AspNet.Mvc.TagHelpers.Internal;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using Microsoft.Framework.Caching.Memory;
+using Microsoft.Framework.Logging;
+using Microsoft.Framework.WebEncoders;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers
+{
+    /// <summary>
+    /// <see cref="ITagHelper"/> implementation targeting &lt;img&gt; elements that supports file versioning.
+    /// </summary>
+    /// <remarks>
+    /// The tag helper won't process for cases with just the 'src' attribute.
+    /// </remarks>
+    [TargetElement("img", Attributes = FileVersionAttributeName)]
+    public class ImageTagHelper : TagHelper
+    {
+        private static readonly string Namespace = typeof(ImageTagHelper).Namespace;
+
+        private const string FileVersionAttributeName = "asp-file-version";
+        private const string SrcAttributeName = "src";
+
+        private FileVersionProvider _fileVersionProvider;
+
+        private static readonly ModeAttributes<Mode>[] ModeDetails = new[] {
+            // Regular src with file version
+            ModeAttributes.Create(Mode.FileVersion, new[] { FileVersionAttributeName }),
+        };
+
+        private enum Mode
```

Comment:
This could be a boolean now. It previously had multiple modes back when this supported src globbing.